contracts/seed/Seed.sol

```diff
@@ -246,44 +216,41 @@ contract Seed {
       * @param _funder           Address of funder to calculate seconds and amount claimable
       * @param _claimAmount      The amount of seed token a users wants to claim.
     */
-    function claim(address _funder, uint256 _claimAmount) public allowedToClaim returns(uint256) {
+    function claim(address _funder, uint256 _claimAmount) external returns(uint256) {
+        require(minimumReached, "Seed: minimum funding amount not met");
+        require(endTime < block.timestamp || maximumReached,"Seed: the distribution has not yet finished");
         uint256 amountClaimable;
 
         amountClaimable = calculateClaim(_funder);
         require(amountClaimable > 0, "Seed: amount claimable is 0");
         require(amountClaimable >= _claimAmount, "Seed: request is greater than claimable amount");
-        uint256 feeAmountOnClaim = (_claimAmount * fee) / 100;
+        uint256 feeAmountOnClaim = (_claimAmount * fee) / PRECISION;
 
-        FunderPortfolio memory tokenFunder = funders[_funder];
-
-        tokenFunder.totalClaimed    += _claimAmount;
-        tokenFunder.feeClaimed      += feeAmountOnClaim;
-        funders[_funder] = tokenFunder;
+        funders[_funder].totalClaimed    += _claimAmount;
 
         seedClaimed += _claimAmount;
-        feeClaimed  += feeAmountOnClaim;
-        require(seedToken.transfer(beneficiary, feeAmountOnClaim), "Seed: cannot transfer to beneficiary");
-        require(seedToken.transfer(_funder, _claimAmount), "Seed: no tokens");
+        require(
+            seedToken.transfer(beneficiary, feeAmountOnClaim) && seedToken.transfer(_funder, _claimAmount),
+            "Seed: seed token transfer failed");
 
         emit TokensClaimed(_funder, _claimAmount, beneficiary, feeAmountOnClaim);
 
-        // fee on the distributed reward collected from admin
-        return (feeAmountOnClaim);
+        return feeAmountOnClaim;
     }
 
     /**
       * @dev         Returns funding tokens to user.
     */
-    function retrieveFundingTokens() public allowedToRetrieve returns(uint256) {
+    function retrieveFundingTokens() external returns(uint256) {
+        require(startTime <= block.timestamp, "Seed: distribution haven't started");
+        require(!minimumReached, "Seed: minimum funding amount met");
         require(funders[msg.sender].fundingAmount > 0, "Seed: zero funding amount");
```

Comment:
seems like it would be more efficient for this line to be below line 249 and read:

```
require(fundingAmount > 0, "Seed: zero funding amount");
```

Reply:
oh yes, will make it more readable.

Reply:
done